Projekty/Projekt1/Kacper_Grzymkowski_Jan_Gaska/Inzynieria Cech, wstępne modelowanie/WUM_Kacper_Grzymkowski_Jan_Gaska_Kamien_Milowy_II.ipynb

```diff
@@ -0,0 +1,2102 @@
+{
```

Comment:
<p>Bardzo dobrze wykonany KM2 - widać umiejętności analityczne i chęć rozłożenia wszystkiego na czynniki pierwsze.</p><p> </p><p>W ramach KM3 warto byłoby wybrać optymalne parametry modeli (na zajęciach zostanie omówione strojenie). Należy uwzględnić kroswalidację.</p><p>Inne możliwe kierunki prac - sugestie:</p><ul><li>analiza drzew decyzyjnych pod kątem ich przycinania (tree pruning) - na zbiorze danych o politykach można to było zrobić na zasadzie prób i błędów  - można jednak do tego zagadnienia podejść w sposób bardziej analityczny</li><li>z uwagi na to, że zbiór jest stosunkowo łatwy a więc tworzone modele biorą pod uwagę niewielką  ilość zmiennych można się zastanowić nad wizualizacją granic decyzyjnych np. w SVM czy kNN</li><li>z uwagi na to, że zbiór jest niezbalansowany można pomyśleć nad zastosowaniem technik upsampling. Innym pomysłem jest zmiana progu odcięcia klas z domyślnej wartości 0.5 na inną</li><li>czy algorytmy do automatyczmej selekcji zmiennych wytypowałyby jako ważne te same zmienne co Wy zidentyfikowaliście?</li></ul><p> </p><p>Uwaga na przyszłość:</p><ul><li>Ciekawa byłaby analiza co jest "wyjątkowego" w obserwacjach, na których finalny model się myli (akurat w przypadku tego zbioru danych w dużym stopniu można to przewidzieć na podstawie EDA). Zajmuje się tym dziedzina Interpretowalnego Uczenia Maszynowego (być może będzie kiedyś okazja wspomnieć o wykorzystywanych algorytmach na zajęciach)</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mini-pw/2021L-WUM/pull/145/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='59'/>